database/sql/instances_test.go

```diff
@@ -0,0 +1,155 @@
+//    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+//    License for the specific language governing permissions and limitations
+//    under the License.
+
+package sql
+
+import (
+	"context"
+	"fmt"
+	dbCommon "garm/database/common"
+	"garm/params"
+	"garm/util"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type InstancesTestFixtures struct {
+	Org       params.Organization
+	Pool      params.Pool
+	Instances []params.Instance
+}
+
+type InstancesTestSuite struct {
+	suite.Suite
+	Store    dbCommon.Store
+	Fixtures *InstancesTestFixtures
+}
+
+func (s *InstancesTestSuite) SetupTest() {
+	// create testing sqlite database
+	db, err := NewSQLDatabase(context.Background(), util.GetTestSqliteDBConfig(s.T()))
+	if err != nil {
+		s.FailNow(fmt.Sprintf("failed to create db connection: %s", err))
+	}
+	s.Store = db
+
+	// create an organization for testing purposes
+	org, err := s.Store.CreateOrganization(context.Background(), "test-org", "test-creds", "test-webhookSecret")
+	if err != nil {
+		s.FailNow(fmt.Sprintf("failed to create org: %s", err))
+	}
+
+	// create an organization pool for testing purposes
+	poolCreateParams := params.CreatePoolParams{
+		ProviderName:   "test-provider",
+		MaxRunners:     4,
+		MinIdleRunners: 2,
+		Image:          "test-image",
+		Flavor:         "test-flavor",
+		OSType:         "linux",
+		Tags:           []string{"self-hosted", "amd64", "linux"},
+	}
+	pool, err := s.Store.CreateOrganizationPool(context.Background(), org.ID, poolCreateParams)
+	if err != nil {
+		s.FailNow(fmt.Sprintf("failed to create org pool: %s", err))
+	}
+
+	// create some instance objects in the database, for testing purposes
+	instances := []params.Instance{}
+	for i := 1; i <= 3; i++ {
+		instance, err := db.CreateInstance(
+			context.Background(),
+			pool.ID,
+			params.CreateInstanceParams{
+				Name:   fmt.Sprintf("test-instance-%v", i),
+				OSType: "linux",
+				OSArch: "amd64",
+				Pool:   pool.ID,
+			},
+		)
+		if err != nil {
+			s.FailNow(fmt.Sprintf("failed to create instance object (test-instance-%v)", i))
+		}
+		instances = append(instances, instance)
+	}
+
+	// setup test fixtures
+	fixtures := &InstancesTestFixtures{
+		Org:       org,
+		Pool:      pool,
+		Instances: instances,
+	}
+	s.Fixtures = fixtures
+}
+
+func (s *InstancesTestSuite) TestCreateInstance() {
+	// setup enviroment for this test
+	instanceName := "test-create-instance"
+	createPoolParams := params.CreateInstanceParams{
+		Name: instanceName,
+		Pool: s.Fixtures.Pool.ID,
+	}
+
+	// call tested function
+	instance, err := s.Store.CreateInstance(context.Background(), s.Fixtures.Pool.ID, createPoolParams)
+
+	// assertions
+	s.Require().Nil(err)
+	storeInstance, err := s.Store.GetInstanceByName(context.Background(), instanceName)
+	if err != nil {
+		s.FailNow(fmt.Sprintf("failed to get instance: %v", err))
+	}
+	s.Require().Equal(storeInstance.Name, instance.Name)
```

Comment:
we should test some more fields like pool ID, provider ID. Some of the fields we explicitly used when creating the entity in the database. Things of this sort. The idea is to test we get back a valid entity after creating it.